outlook/src/taskpane/components/CollapseSection/CollapseSection.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from "react";
+import {faChevronDown, faChevronRight, faPlus} from "@fortawesome/free-solid-svg-icons";
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import "./CollapseSection.css";
+import {ReactElement} from "react";
+
+
+type CollapseSectionProps = {
+    title: string;
+    isCollapsed: boolean;
+    onCollapseButtonClick: () => void;
+    hasAddButton?: boolean;
+    onAddButtonClick?: () => void;
+    hideCollapseButton?: boolean;
+    children: ReactElement;
+}
+
+type CollapseSectionHeaderState = {
```

Comment:
What is the purpose of this void CollapseSectionHeaderState ?

Reply:
I found this in the old codebase, it is used to provide the required type argument for the React Component class, since it is mandatory to specify a state type even if it is an "empty" one. After taking a deeper look into this, I got rid of it and managed to use {} directly inside the class declaration which is much cleaner I think.